elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/impl/ElasticsearchIndexManager.java

```diff
@@ -115,7 +147,15 @@ public void setSearchFactory(ExtendedSearchIntegrator boundSearchIntegrator) {
 		this.searchIntegrator = boundSearchIntegrator;
 		this.clientReference = searchIntegrator.getServiceManager().requestReference( JestClient.class );
 
-		initializeIndex();
+		// for dynamic sharding postInitialize() will not be invoked by the builder, so doing it here ourselves
+		if ( postInitialize != null ) {
+			try {
+				postInitialize.call();
+			}
+			catch (Exception e) {
+				throw new SearchException( "Index manager initialization failed", e );
```

Comment:
What I head in mind was to `Logger`-ify everything once we finalize all this work and the dust has settled.

As long as things are in flow, I noticed one moves forth and back with creating these logger methods which seems like a waste of time.